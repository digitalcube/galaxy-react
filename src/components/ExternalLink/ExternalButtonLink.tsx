import React, { FC, ReactNode } from 'react';
import { Button, ButtonProps } from 'reactstrap';

export type ExternalButtonLinkProps = {
  icon?: ReactNode;
} & Omit<ButtonProps, 'tag'>;
export const ExternalButtonLink: FC<ExternalButtonLinkProps> = ({
  icon,
  children,
  role,
  color = 'primary',
  ...buttonProps
}) => {
  const className = [
    'px-3 py-0 rounded btn-bordered btn-smaller d-inline-flex align-items-center font-weight-bold',
    buttonProps.className,
  ]
    .filter(Boolean)
    .join(' ');
  const rel = [
    'noopener',
    buttonProps.rel,
    /getshifter.io/.test(buttonProps.href) ? '' : 'noreferrer',
  ]
    .filter(Boolean)
    .join(' ');
  return (
    <Button
      {...buttonProps}
      color={color}
      className={className}
      rel={rel}
      target="_blank"
      tag="a"
    >
      {children}
      {icon ? (
        <i className="pl-2 larger-icon" aria-hidden="true">
          {icon}
        </i>
      ) : null}
    </Button>
  );
};
