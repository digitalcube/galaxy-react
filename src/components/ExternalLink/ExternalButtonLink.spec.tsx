import { render } from '@testing-library/react';
import React from 'react';
import { FaCircle } from 'react-icons/fa';
import { ExternalButtonLink } from './ExternalButtonLink';

describe('ExternalButtonLink', () => {
  it.each([
    [
      {
        href: 'https://getshifter.io',
        children: 'Link',
      },
    ],
    [
      {
        href: 'https://example.com',
        children: 'Link',
      },
    ],
    [
      {
        href: 'https://getshifter.io',
        children: 'Link',
        icon: <FaCircle className="d-block" />,
      },
    ],
    [
      {
        href: 'https://example.com',
        children: 'Link',
        icon: <FaCircle className="d-block" />,
      },
    ],
  ])('given %o, should match snapshot', (args) => {
    const { asFragment } = render(<ExternalButtonLink {...(args as any)} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
