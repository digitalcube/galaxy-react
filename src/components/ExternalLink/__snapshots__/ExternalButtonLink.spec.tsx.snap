// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalButtonLink given {
  href: 'https://example.com',
  children: 'Link',
  icon: {
    '$$typeof': Symbol(react.element),
    type: <ref *1> [Function: FaCircle] {
      [length]: 1,
      [name]: 'FaCircle',
      [arguments]: null,
      [caller]: null,
      [prototype]: { [constructor]: [Circular *1] }
    },
    key: null,
    ref: null,
    props: { className: 'd-block' },
    _owner: null,
    _store: { [validated]: false },
    [_self]: null,
    [_source]: null
  }
}, should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="px-3 py-0 rounded btn-bordered btn-smaller d-inline-flex align-items-center font-weight-bold btn btn-secondary"
    href="https://example.com"
    rel="noopenernoreferrer"
    target="_blank"
  >
    Link
    <i
      aria-hidden="true"
      class="pl-2 larger-icon"
    >
      <svg
        class="d-block"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </i>
  </a>
</DocumentFragment>
`;

exports[`ExternalButtonLink given {
  href: 'https://getshifter.io',
  children: 'Link',
  icon: {
    '$$typeof': Symbol(react.element),
    type: <ref *1> [Function: FaCircle] {
      [length]: 1,
      [name]: 'FaCircle',
      [arguments]: null,
      [caller]: null,
      [prototype]: { [constructor]: [Circular *1] }
    },
    key: null,
    ref: null,
    props: { className: 'd-block' },
    _owner: null,
    _store: { [validated]: false },
    [_self]: null,
    [_source]: null
  }
}, should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="px-3 py-0 rounded btn-bordered btn-smaller d-inline-flex align-items-center font-weight-bold btn btn-secondary"
    href="https://getshifter.io"
    rel="noopener"
    target="_blank"
  >
    Link
    <i
      aria-hidden="true"
      class="pl-2 larger-icon"
    >
      <svg
        class="d-block"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
        />
      </svg>
    </i>
  </a>
</DocumentFragment>
`;

exports[`ExternalButtonLink given { href: 'https://example.com', children: 'Link' }, should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="px-3 py-0 rounded btn-bordered btn-smaller d-inline-flex align-items-center font-weight-bold btn btn-secondary"
    href="https://example.com"
    rel="noopenernoreferrer"
    target="_blank"
  >
    Link
  </a>
</DocumentFragment>
`;

exports[`ExternalButtonLink given { href: 'https://getshifter.io', children: 'Link' }, should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="px-3 py-0 rounded btn-bordered btn-smaller d-inline-flex align-items-center font-weight-bold btn btn-secondary"
    href="https://getshifter.io"
    rel="noopener"
    target="_blank"
  >
    Link
  </a>
</DocumentFragment>
`;
